packages/instantsearch-ui-components/src/components/autocomplete/AutocompleteIndex.tsx

```diff
@@ -0,0 +1,49 @@
+/** @jsx createElement */
+
+import { cx } from '../../lib/cx';
+
+import type { Renderer } from '../../types';
+
+export type AutocompleteIndexProps<
+  T = { objectID: string } & Record<string, unknown>
+> = {
+  items: T[];
+  ItemComponent: (item: T) => JSX.Element;
+  classNames?: Partial<AutocompleteIndexClassNames>;
+};
+
+export type AutocompleteIndexClassNames = {
+  /**
+   * Class names to apply to the root element
+   **/
+  root: string | string[];
+  /**
+   * Class names to apply to the list element
+   */
+  list: string | string[];
+  /**
+   * Class names to apply to each item element
+   */
+  item: string | string[];
+};
+
+export function createAutocompleteIndexComponent({ createElement }: Renderer) {
+  return function AutocompleteIndex(userProps: AutocompleteIndexProps) {
+    const { items, ItemComponent, classNames = {} } = userProps;
+
+    return (
+      <div className={cx('ais-AutocompleteIndex', classNames.root)}>
+        <ul className={cx('ais-AutocompleteIndexList', classNames.list)}>
```

Comment:
I think there probably (later) should be only one list, as you're navigating across all lists with the keyboard. However there's an argument to be made for separate lists for each index, as that allows us to have header/footer per index.